required/tools/build.lua

```diff
@@ -38,3 +38,28 @@ checkconfigs = {"build","config-TU","config-legacy","config-search"}
 
 -- Load the common settings for the LaTeX2e repo
 dofile (maindir .. "/build-config.lua")
+
+-- special code to handle .tex
+
+function unpack(sources, sourcedirs)
+  local errorlevel = dep_install(unpackdeps)
+  if errorlevel ~= 0 then
+    return errorlevel
+  end
+  errorlevel = bundleunpack(sourcedirs, sources)
+  if errorlevel ~= 0 then
+    return errorlevel
+  end
+  texio.write(" * Renaming rename-to-empty-base.tex to .tex")
+  errorlevel = ren(unpackdir,"rename-to-empty-base.tex",".tex")
+  if errorlevel ~= 0 then
+    return errorlevel
+  end
+  for _,i in ipairs(installfiles) do
+    errorlevel = cp(i, unpackdir, localdir)
+    if errorlevel ~= 0 then
+      return errorlevel
+    end
+  end
+  return 0
+end
```

Comment:
This patched `unpack()` is not used by l3build, as the reference to the original `unpack()`, not `"unpack"` the function name in string is stored in the `target_list` table. See [`stdmain.lua` from l3build](https://github.com/latex3/l3build/blob/d888be068514d7131469d4cbab671b2cc8dc7fd7/l3build-stdmain.lua#L137-L142)).

That's why the check failed on test `required/tools/testfiles/github-0222-input.lvt` which tried to load `.tex` file, see https://github.com/latex3/latex2e/actions/runs/10834966806/job/30065438521.

Updating the binding in `target_list` does the trick, though in the long term I think `l3build` could provide a new hook `unpack_hook()`.

```suggestion
end

-- update function binding
target_list.unpack.func = unpack
```

Reply:
Ah it was used when I called l3build install directly in tools directory, but not from the top level,  thanks for the fix

Reply:
I did look for a hook first:-)

Reply:
@muzimuzhi  hmm this passes locally  but is still failing in this PR

Reply:
check calls bundleunpack directly not unpack, so I think I need to hook there

Reply:
Sorry the check still failed at `github-0222-input.lvt`. Digging...

Reply:
@muzimuzhi  yes see what happens now (hooking to bundleunpack and guarded just to affect tools)

Reply:
@muzimuzhi  thanks for the help, seems it all passes now

Reply:
> check calls bundleunpack directly not unpack, so I think I need to hook there

Ah yes. This is tricky.

(I didn't see your replies when typing in my previous comment.)